abi/codegen.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package abi
+
+import (
+	"fmt"
+	"go/format"
+	"strings"
+	"unicode"
+
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+func GenerateGoStructs(abi ABI, packageName string) (string, error) {
+	var sb strings.Builder
+
+	sb.WriteString(fmt.Sprintf("package %s\n\n", packageName))
+	sb.WriteString("import (\n\t\"github.com/ava-labs/hypersdk/codec\"\n)\n\n")
+
+	processed := set.Set[string]{}
+
+	for _, typ := range abi.Types {
+		if processed.Contains(typ.Name) {
+			continue
+		}
+		processed.Add(typ.Name)
+
+		sb.WriteString(fmt.Sprintf("type %s struct {\n", typ.Name))
+		for _, field := range typ.Fields {
+			fieldNameUpperCase := strings.ToUpper(field.Name[0:1]) + field.Name[1:]
```

Comment:
Hmm, I don't think so. It's a string, not []byte

Reply:
Golang will grab the first byte instead of first character if you index into a string this way. Golang be weird, but as discussed offline I think this is ok to leave as is.